test/unit/models.spec.js

```diff
@@ -0,0 +1,181 @@
+    });
+
+    it('should return [block, article] if they both have trickle configurations', () => {
+      const article = new MockModel({
+        _type: 'article',
+        _trickle: { _isEnabled: true, _onChildren: false }
+      });
+      const block = new MockModel({
+        _type: 'block',
+        _parent: article,
+        _trickle: { _isEnabled: true }
+      });
+
+      expect(getModelInheritanceChain(block)).toStrictEqual([block, article]);
+    });
+
+    it('should omit the block if it explicitly requires inheritance', () => {
+      const article = new MockModel({
+        _type: 'article',
+        _trickle: { _isEnabled: true, _onChildren: false }
+      });
+      const block = new MockModel({
+        _type: 'block',
+        _parent: article,
+        _trickle: { _isEnabled: true, _isInherited: true }
+      });
+
+      expect(getModelInheritanceChain(block)).toStrictEqual([article]);
+    });
+  });
+});
+
+describe('getModelConfig', () => {
+
+  it('should return null if trickle is not enabled via the model or via inheritance', () => {
+    const article = new MockModel({
+      _type: 'article',
+      _trickle: { _isEnabled: false, _onChildren: true }
+    });
+    const block = new MockModel({
+      _type: 'block',
+      _parent: article
+    });
+
+    const config = getModelConfig(block);
+
+    expect(config).toBeNull();
+  });
+
+  it('should return null if the derived configuration is not enabled', () => {
+    const article = new MockModel({
+      _type: 'article',
+      _trickle: { _isEnabled: true }
+    });
+    const block = new MockModel({
+      _type: 'block',
+      _parent: article,
+      _trickle: { _isEnabled: false }
+    });
+
+    const config = getModelConfig(block);
+
+    expect(config).toBeNull();
+  });
+
+  describe('when given a block', () => {
+
+    it('should override inherited configuration with defaults and then own properties', () => {
+      const article = new MockModel({
+        _type: 'article',
+        _trickle: { _isEnabled: false, _button: { _isEnabled: false } }
+      });
+      const block = new MockModel({
+        _type: 'block',
+        _parent: article,
+        _trickle: { _isEnabled: true }
+      });
+
+      const config = getModelConfig(block);
+
+      expect(config._button._isEnabled).toBeTruthy();
+      expect(config._isEnabled).toBeTruthy();
+    });
+
+    it('should inherit configuration from the article ', () => {
+      const article = new MockModel({
+        _type: 'article',
+        _trickle: { _isEnabled: false, _scrollDuration: 250 }
+      });
+      const block = new MockModel({
+        _type: 'block',
+        _parent: article,
+        _trickle: { _isEnabled: true }
+      });
+
+      const config = getModelConfig(block);
+      expect(config._scrollDuration).toStrictEqual(250);
+    });
+  });
+});
+
+describe('getModelContainer', () => {
+
+  // TODO: getModelContainer should ignore _onChildren:true on block (per README) but does not
```

Comment:
I believe these lines do force a false state on the _onChildren unless article level
https://github.com/adaptlearning/adapt-contrib-trickle/blob/master/js/models.js#L89-L91

Reply:
I noticed if `_onChildren:true` was applied to the block's trickle config (in the JSON) this function would return the block when it should return the article. I did some hasty testing but didn't notice any problems as a result of doing this.

Reply:
Is there a pr needed to fix this in trickle?

Reply:
I think it'd be good to, if only for correctness. I'll work in some more tests and see if anything else comes up and raise PRs together if you're both happy with that?